src/Reason.ml

```diff
@@ -13,25 +13,91 @@
 
 (** Type of Reason binary serialized data, which must agree with the type
     used by the implementation of `refmt`. *)
-type 'a reason_data =
-  string * string * 'a * (string * int * Location.t) list * bool * bool
+
+type category = int
+
+(* | EndOfLine | SingleLine | Regular *)
+
+type comment = {location: Location.t; category: category; text: string}
+
+type 'a reason_data = string * string * 'a * comment list * bool * bool
+
+(* copy and adapted from ocaml-migrate_parsetree *)
+type ast =
+  | Impl :
+      (module Migrate_parsetree_versions.OCaml_version
+         with type Ast.Parsetree.structure = 'concrete)
+      * 'concrete
+      -> ast
+  | Intf :
+      (module Migrate_parsetree_versions.OCaml_version
+         with type Ast.Parsetree.signature = 'concrete)
+      * 'concrete
+      -> ast
+
+let find_magic magic x =
+  let rec loop = function
+    | [] ->
+        let prefix = String.sub magic ~pos:0 ~len:9 in
+        if
+          String.equal prefix
+            (String.sub Ast_402.Config.ast_impl_magic_number ~pos:0 ~len:9)
```

Comment:
402?

Reply:
the version doesn't matter, we only care about the prefix.